paddle/fluid/distributed/table/weighted_sampler.h

```diff
@@ -16,46 +16,40 @@
 #include <ctime>
 #include <unordered_map>
 #include <vector>
+#include "paddle/fluid/distributed/table/graph_edge.h"
 namespace paddle {
 namespace distributed {
 
-class WeightedObject {
- public:
-  WeightedObject() {}
-  virtual ~WeightedObject() {}
-  virtual uint64_t get_id() = 0;
-  virtual float get_weight() = 0;
-};
-
 class Sampler {
 public:
   virtual ~Sampler() {}
-  virtual void build(std::vector<WeightedObject*>* edges) = 0;
-  virtual std::vector<WeightedObject *> sample_k(int k) = 0;
+  virtual void build(GraphEdgeBlob* edges) = 0;
+  virtual std::vector<int> sample_k(int k) = 0;
 };
 
 class RandomSampler: public Sampler {
 public:
   virtual ~RandomSampler() {}
-  virtual void build(std::vector<WeightedObject*>* edges);
-  virtual std::vector<WeightedObject *> sample_k(int k);
-  std::vector<WeightedObject*>* edges;
+  virtual void build(GraphEdgeBlob* edges);
+  virtual std::vector<int> sample_k(int k);
+  GraphEdgeBlob* edges;
 };
 
 class WeightedSampler: public Sampler {
  public:
   WeightedSampler();
   virtual ~WeightedSampler();
   WeightedSampler *left, *right;
-  WeightedObject *object;
-  int count;
   float weight;
-  virtual void build(std::vector<WeightedObject*>* edges);
-  virtual void build_one(WeightedObject **v, int start, int end);
-  virtual std::vector<WeightedObject *> sample_k(int k);
+  int count;
+  int idx;
+  WeightedGraphEdgeBlob * edges;
```

Comment:
I assume we don't have to specify it's a WeightedGraphEdgeBlob* object,
since GraphEdgeBlob also has get_weight function, our application could still run when setting edges type to  GraphEdgeBlob *, but by doing this, our application becomes more robust.
(when actually used, its actual type is WeightedGraphEdgeBlob*, but it's type could be declared as GraphEdgeBlob* here.)

Reply:
Ok, just fix it.